### PR TITLE
Search Bar Changes

### PR DESCRIPTION
**Pull request by carkom**
Looking to make the search bar a bit more user friendly while also avoiding it being too loud.

It's been suggested that this also be applied to all search bars (payees, schedules, etc.) - which I'm fine with.

### CODE REVIEW

**Comment by biohzrddd** [comment]
This looks good but still has the bug where the X goes outside of the search bar on very narrow windows (just before transitioning to mobile). Would definitely like to see the styles made common across payee/schedule with the exception of the account view being a bit more "background" on accounts view. 

image

**Comment by j-f1** [comment]
@MatissJanis interested to see what you think of this at a product level!
Screenshot_2023-07-28 09 18 28

**Comment by MatissJanis** [comment]
Cool, I like this. 👍

File: packages/desktop-client/src/style/themes/light.ts [hunk 1]
```diff
@@ -124,11 +124,13 @@ export const formInputBackgroundSelection = colorPalette.purple500;
 export const formInputBorder = colorPalette.navy300;
 export const formInputTextReadOnlySelection = colorPalette.navy50;
 export const formInputBorderSelected = colorPalette.purple500;
+export const altformInputBorderSelected = colorPalette.blue500;
```
Comment (j-f1) [inline packages/desktop-client/src/style/themes/light.ts:127]: Where is the non-`alt` version of this color used? I would be in favor of changing the input to have a purple border when focused, or change the regular color back to blue and using that.

File: packages/desktop-client/src/components/common/Search.tsx [hunk 2]
```diff
@@ -30,13 +30,8 @@ export default function Search({
...
         borderColor: isInModal ? null : 'transparent',
-        backgroundColor: isInModal ? null : colors.n11,
-        ':focus': isInModal
-          ? null
-          : {
-              backgroundColor: 'white',
-              '::placeholder': { color: colors.n8 },
-            },
+        backgroundColor: isInModal ? null : theme.formInputBackground,
+        color: 'inherit',
```
Comment (j-f1) [inline packages/desktop-client/src/components/common/Search.tsx:34]: Is this override necessary?

**Comment by carkom** [comment]
> This looks good but still has the bug where the X goes outside of the search bar on very narrow windows (just before transitioning to mobile). 
Fixed!
> Would definitely like to see the styles made common across payee/schedule with the exception of the account view being a bit more "background" on accounts view.
I've normalized them all (payees, schedules, rules, accounts). Let me know if I've missed any. 

I'm also happy to adjust any of them if there is preference. Keeping in mind the theme's we can't customize them too much as we need to stay within a few options to make the light/dark work properly.

File: packages/desktop-client/src/style/themes/light.ts (hunk 1 above)
Comment (carkom) [reply packages/desktop-client/src/style/themes/light.ts:127]: I'm fine with either way. I was actually going to ask about this. Does anyone have a preference? 

@MatissJanis do you care if it's a blue border or a purpule border when in focus?

File: packages/desktop-client/src/components/common/Search.tsx (hunk 2 above)
Comment (carkom) [reply packages/desktop-client/src/components/common/Search.tsx:35]: This was there for testing, thanks for catching it. Removed.

File: packages/desktop-client/src/style/themes/light.ts (hunk 1 above)
Comment (MatissJanis) [reply packages/desktop-client/src/style/themes/light.ts:127]: I think purple makes more sense as it's our primary color.

**Comment by j-f1** [comment]
Can you make them either able to dynamically shrink as needed or just have an explicit smaller size on narrower screens? They’re messing a bit with the layout:

image
image

**Comment by carkom** [comment]
> Can you make them either able to dynamically shrink as needed or just have an explicit smaller size on narrower screens? They’re messing a bit with the layout:

The search bar was always dynamically set to flex. I changed the items getting messed with to flexshrink=0 and that seems to have fixed it. Let me know if it hasn't.

**Comment by carkom** [comment]
@j-f1 , I didn't want to mess with the theme colors but some of them seemed a bit odd - I'm not sure how settled these colors are or what I might be messing up elsewhere in the app. eg. formInputBackgroundSelected being purple in every mode felt really loud. I changed a few to match the existing functionality of the app. I won't be upset if we need to adjust any of these.

**Comment by biohzrddd** [comment]
> @j-f1 , I didn't want to mess with the theme colors but some of them seemed a bit odd - I'm not sure how settled these colors are or what I might be messing up elsewhere in the app. eg. formInputBackgroundSelected being purple in every mode felt really loud. I changed a few to match the existing functionality of the app. I won't be upset if we need to adjust any of these.

@carkom take a look at #922 for the fully implemented colors with the caveat that some components have changed since then. If you still think colors look out of place there, let me know.

**Comment by j-f1** [comment]
Looks like it’s just white/black in #922, not sure how it got messed up in `master`.

**Comment by carkom** [comment]
I think this is all good if there's no other changes or suggestions.

@MatissJanis

File: packages/desktop-client/src/components/payees/index.js [hunk 3]
```diff
@@ -482,7 +482,11 @@ export const ManagePayees = forwardRef(
             padding: '0 10px 5px',
           }}
         >
-          <View>
+          <View
+            style={{
+              flexShrink: 0,
+            }}
+          >
```
Comment (j-f1) [inline packages/desktop-client/src/components/payees/index.js:489]: Small nit:

```suggestion
          <View style={{ flexShrink: 0 }}>
```

**Review by j-f1 — COMMENTED** [review]
Small nit but otherwise looks good to me!

**Comment by MatissJanis** [comment]
Just tested once more: I think this is great. Works also on smaller screen sizes. :)

**Comment by j-f1** [comment]
yeah, this is a huge improvement to all the search bars!

**Review by j-f1 — APPROVED** [review]
Thanks!

**Review by j-f1 — APPROVED** [review]
Excellent style cleanup!

**Pull request merged**